### PR TITLE
[#983] improvement(tez): Optimize tez client delivery configuration

### PR DESCRIPTION
**Pull request by zhengchenyu**
### What changes were proposed in this pull request?

Three improvement about configuration will be done in this issue.

- 1 For now, tez client use rss_conf.xml to delivery configuration. As https://github.com/apache/incubator-uniffle/pull/966 is applied, we can delivery configuration by edge conf.
- 2 delivery dynamic configuration from coordinator, then override the tez client configuration.
- 3 delivery configuration from client side.

### Why are the changes needed?

- 1. rss_conf.xml is unnecessary.
- 2. dynamic configuration from coordinator are not applied.
- 3. config in client side can not delivery to input/ouput.

### How was this patch tested?

integration test, unit test, test in yarn cluster, test in tez local mode.

### CODE REVIEW

**Comment by jerqi** [comment]
> ### What changes were proposed in this pull request?
> Three improvement about configuration will be done in this issue.
> 
> * 1 For now, tez client use rss_conf.xml to delivery configuration. As [[#965] feat(tez): support remote shuffle for tez framework #966](https://github.com/apache/incubator-uniffle/pull/966) is applied, we can delivery configuration by edge conf.
> * 2 delivery dynamic configuration from shuffle server, then override the tez client configuration.
> * 3 delivery configuration from client side.
> 
> ### Why are the changes needed?
> * 1. rss_conf.xml is unnecessary.
> * 2. dynamic configuration from shuffle server are not applied.

Should `shuffle server` be changed to `coordinator`?

**Comment by zhengchenyu** [comment]
> > ### Why are the changes needed?
> > 
> > * 1. rss_conf.xml is unnecessary.
> > * 2. dynamic configuration from shuffle server are not applied.
> 
> Should `shuffle server` be changed to `coordinator`?

OK! modification is Done!

File: client-tez/src/main/java/org/apache/tez/dag/app/RssDAGAppMaster.java [hunk 1]
```diff
@@ -395,36 +318,70 @@ public void run() {
...
+    
+    DagInitialCallback(RssDAGAppMaster appMaster) {
+      this.appMaster = appMaster;
+    } 
+
     @Override
     public void onStateChanged(DAGImpl dag, DAGState dagState) {
       try {
+        // get rss config from client
+        Configuration filterRssConf = RssTezUtils.filterRssConf(appMaster.getConfig());
```
Comment (baitian77) [inline client-tez/src/main/java/org/apache/tez/dag/app/RssDAGAppMaster.java:338]: 不仅仅是RSS的参数需要传递，还有tez原有的io input/ouput getConfigurationKeySet()里的参数都需要传递，例如TEZ_RUNTIME_IO_FILE_BUFFER_SIZE

File: client-tez/src/main/java/org/apache/tez/common/RssTezUtils.java [hunk 2]
```diff
@@ -393,4 +393,40 @@ public static String replaceRssInputClassName(String className) {
...
+      if (StringUtils.isEmpty(conf.get(tezConfKey, ""))
+          || RssTezConfig.RSS_MANDATORY_CLUSTER_CONF.contains(tezConfKey)) {
+        LOG.warn("Use conf dynamic conf {} = {}", tezConfKey, tezConfVal);
+        conf.set(tezConfKey, tezConfVal);
+      }
+    }
+  }
+
+  public static Configuration filterRssConf(Configuration extraConf) {
+    Configuration conf = new Configuration();
```
Comment (baitian77) [inline client-tez/src/main/java/org/apache/tez/common/RssTezUtils.java:423]: 先清空默认配置conf.clear()

Comment (zhengchenyu) [reply client-tez/src/main/java/org/apache/tez/common/RssTezUtils.java:423]: Great catch! I do it like below.  It is more clear.
```
Configuration conf = new Configuration(false);
```

File: client-tez/src/main/java/org/apache/tez/dag/app/RssDAGAppMaster.java (hunk 1 above)
Comment (zhengchenyu) [reply client-tez/src/main/java/org/apache/tez/dag/app/RssDAGAppMaster.java:338]: the configs in getConfigurationKeySet have delivered. 
The process of filter with getConfigurationKeySet happen when call setFromConfiguration in client side. 
I will add a unit test to prove it.

Comment (baitian77) [reply client-tez/src/main/java/org/apache/tez/dag/app/RssDAGAppMaster.java:338]: Yes，you are right, here we no need to delivered configs in getConfigurationKeySet

File: client-tez/src/main/java/org/apache/tez/common/RssTezUtils.java (hunk 2 above)
Comment (baitian77) [reply client-tez/src/main/java/org/apache/tez/common/RssTezUtils.java]: I think still need to call conf.clear(). Even if  Configuration(false),  will still load some configurations

Comment (baitian77) [reply client-tez/src/main/java/org/apache/tez/common/RssTezUtils.java]: After test,  it works without clear

**Review by baitian77 — APPROVED** [review]

**Review by jerqi — APPROVED** [review]
LGTM, thanks @zhengchenyu @baitian77

**Pull request merged**